### PR TITLE
better error message for cleanDirectBuffer

### PR DESCRIPTION
**Pull request by siyangy**
https://tachyon.atlassian.net/browse/TACHYON-1442

Only log the error message when the buffer is supposed to have a non-null Cleaner. Otherwise when the buffer is empty, the Cleaner is actually set as null during initialization.

### CODE REVIEW

**Comment by yupeng9** [comment]
Thanks for improving the error logging! Could you add a comment in the source code on when the cleaner is expected to be null?

File: common/src/main/java/tachyon/util/io/BufferUtils.java [hunk 1]
```diff
@@ -73,7 +73,9 @@ public static void cleanDirectBuffer(ByteBuffer buffer) {
       }
       final Object cleaner = sByteBufferCleanerMethod.invoke(buffer);
       if (cleaner == null) {
-        LOG.error("Failed to get cleaner for ByteBuffer");
+        if (buffer.capacity() > 0) {
```
Comment (calvinjia) [inline common/src/main/java/tachyon/util/io/BufferUtils.java]: Could we use `if (cleaner == null && buffer.capacity() > 0)`?

Comment (apc999) [inline common/src/main/java/tachyon/util/io/BufferUtils.java]: i guess that is different from what @siyangy is having here.
In the current code, as long as cleaner is null, we return.

**Comment by apc999** [comment]
thanks for the patch! @yupeng9 has a good point in adding some more comment. After that, I will LGTM this PR.

**Comment by yupeng9** [comment]
LGTM

**Pull request merged**